### PR TITLE
refactor: replace `deferred()` from `std/async` with `Promise.withResolvers()`

### LINKED ISSUE #21041: Migrate tests to use "Promise.withResolvers()"
When https://github.com/denoland/deno/pull/21036 lands the new `Promise.withResolvers()` API will be available. We should migrate `cli/tests/` to use this new API instead of `deferred` from `std/async/mod.ts`.

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #21041

### CODE REVIEW

File: cli/tests/testdata/run/tls_connecttls.js [hunk 1]
```diff
@@ -9,7 +8,7 @@ import { TextProtoReader } from "./textproto.ts";
 const encoder = new TextEncoder();
 const decoder = new TextDecoder();
 
-const resolvable = deferred();
+const { promise, resolve } = Promise.withResolvers();
```
Comment (bartlomieju) [inline cli/tests/testdata/run/tls_connecttls.js:11]: Maybe let's not destructure bindings here and name it `resolveable`?

Comment (iuioiua) [reply cli/tests/testdata/run/tls_connecttls.js:11]: No strong opinions against this but how come?

Comment (bartlomieju) [reply cli/tests/testdata/run/tls_connecttls.js:11]: In some tests you'll need to create multiple deferreds, I think it's a bit easier to understand if you have `resolvable1` and `resolvable2` than `promise1`, `resolve1`, reject1`, `promise2`, `resolve2`, `reject2`.

Comment (iuioiua) [reply cli/tests/testdata/run/tls_connecttls.js:11]: I see. I've only destructured in tests that contain 1 occurrence of `Promise.withResolvers()`. The rest adhere to your suggestion.

**Comment by bartlomieju** [comment]
Oooops, did I screw up the merge order? 🤔 @iuioiua could you please rebase this PR?

**Comment by iuioiua** [comment]
No, it was inevitable, with both PRs queued at the same time. Rebased.

**Review by bartlomieju — APPROVED** [review]
LGTM, great refactor. Thank you Asher!

**Pull request merged**